Robus/src/msgAlloc.c

```diff
@@ -0,0 +1,163 @@
+ ******************************************************************************/
+void MsgAlloc_ValidHeader()
+{
+    /******** End the message **********
+     * - Put it on the module table
+     */
+    /******** Prepare the allocator to get data  *********
+     * - Be sure that the end of msg_buffer is after data_ptr + header_t.size
+     * - If the end of buffer is before the end of the message :
+     *     - check if there is no msg between &msg_buffer[0] and &msg_buffer[0] + sizeof(header_t) + header_t.size
+     *     - If there is one remove it and set the proper varible into the module
+     *     - Copy the header at the begining of msg_buffer
+     *     - move current_msg to &msg_buffer[0]
+     *     - move data_ptr to current_msg + sizeof(header_t)
+     * - If not 
+     *     - check if there is no msg between &msg_buffer[0] and &msg_buffer[0] + sizeof(header_t) + header_t.size
+     *     - If there is one remove it and set the proper varible into the module
+     */
+}
+
+/******************************************************************************
+ * @brief Finish and valid the current message and prepare the allocator to get the next one
+ * @param None
+ * @return None
+ ******************************************************************************/
+void MsgAlloc_EndMsg()
+{
+    /******** End the message **********
+     * - MsgAlloc_FindAllocSlot()
+     * - fill alloc_table[alloc_id] struct
+     */
+    /******** Prepare the next msg *********
+     * - be sure that the end of msg_buffer is after data_ptr + sizeof(header_t)
+     * - update the current_msg = data_ptr
+     * - check if thers is no msg between current_msg and current_msg + sizeof(header_t)
+     * - If there is one remove it
+     */
+}
+
+/******************************************************************************
+ * @brief write a byte into the current message.
+ * @param uint8_t data to write in the allocator
+ * @return None
+ ******************************************************************************/
+uint8_t MsgAlloc_SetData(uint8_t data)
+{
+    /******** Write data  *********
+     * - copy data into *data_ptr
+     * - move data_ptr forward data_ptr++
+     */
+}
+
+/******************************************************************************
+ * @brief return the current message
+ * @param None
+ * @return msg_t* pointer to the current message
+ ******************************************************************************/
+msg_t *MsgAlloc_GetCurrentMsg(void)
+{
+    return current_msg;
+}
+
+/******************************************************************************
+ * @brief get back alloc_table and alloc_id
+ * @param
+ *     msg_allocator_t *_alloc_table : The table linking modules and messages
+ *     uint16_t *_alloc_id : The last writen msg
+ * @return None
+ ******************************************************************************/
+void MsgAlloc_GetAllocVars(msg_allocator_t *_alloc_table, uint16_t *_alloc_id)
```

Comment:
@Salem-Luos : Do you think this function is a good thing or not ?

I use it to share `alloc_table` and `alloc_id` vars. I could do it using *extern*.

This function cost flash, time at init, and ask luos to include msgAlloc.h.